docs/data/data-grid/state/RestoreStateApiRef.js

```diff
@@ -0,0 +1,358 @@
+};
+
+const NewViewListButton = (props) => {
+  const { label, onLabelChange, onSubmit, isValid } = props;
+  const [isAddingView, setIsAddingView] = React.useState(false);
+
+  const handleSubmitForm = (e) => {
+    onSubmit();
+    setIsAddingView(false);
+    e.preventDefault();
+  };
+
+  return (
+    <React.Fragment>
+      <Button
+        endIcon={<AddIcon />}
+        size="small"
+        onClick={() => setIsAddingView(true)}
+      >
+        Save current view
+      </Button>
+      <Dialog onClose={() => setIsAddingView(false)} open={isAddingView}>
+        <form onSubmit={handleSubmitForm}>
+          <DialogTitle>New custom view</DialogTitle>
+          <DialogContent>
+            <TextField
+              autoFocus
+              value={label}
+              onChange={onLabelChange}
+              margin="dense"
+              size="small"
+              label="Custom view label"
+              variant="standard"
+              fullWidth
+            />
+          </DialogContent>
+          <DialogActions>
+            <Button type="button" onClick={() => setIsAddingView(false)}>
+              Cancel
+            </Button>
+            <Button type="submit" disabled={!isValid}>
+              Create view
+            </Button>
+          </DialogActions>
+        </form>
+      </Dialog>
+    </React.Fragment>
+  );
+};
+
+NewViewListButton.propTypes = {
+  isValid: PropTypes.bool.isRequired,
+  label: PropTypes.string.isRequired,
+  onLabelChange: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+};
+
+const CustomToolbar = () => {
+  const apiRef = useGridApiContext();
+  const [state, dispatch] = React.useReducer(demoReducer, DEMO_INITIAL_STATE);
+
+  const createNewView = () => {
+    dispatch({
+      type: 'createView',
+      value: apiRef.current.exportState(),
+    });
+  };
+
+  const handleNewViewLabelChange = (e) => {
+    dispatch({ type: 'setNewViewLabel', label: e.target.value });
+  };
+
+  const handleDeleteView = React.useCallback((viewId) => {
+    dispatch({ type: 'deleteView', id: viewId });
+  }, []);
+
+  const handleSetActiveView = (viewId) => {
+    apiRef.current.restoreState(state.views[viewId].value);
+    dispatch({ type: 'setActiveView', id: viewId });
+  };
+
+  const handlePopperAnchorClick = (event) => {
+    dispatch({ type: 'togglePopper', element: event.currentTarget });
+    event.stopPropagation();
+  };
+
+  const handleClosePopper = () => {
+    dispatch({ type: 'closePopper' });
+  };
+
+  const isNewViewLabelValid = React.useMemo(() => {
+    if (state.newViewLabel.length === 0) {
+      return false;
+    }
+
+    return Object.values(state.views).every(
+      (view) => view.label !== state.newViewLabel,
+    );
+  }, [state.views, state.newViewLabel]);
+
+  const canBeMenuOpened = state.isMenuOpened && Boolean(state.menuAnchorEl);
+  const popperId = canBeMenuOpened ? 'transition-popper' : undefined;
+
+  return (
+    <GridToolbarContainer style={{ height: 48 }}>
```

Comment:
The toolbar has an additional space on the top.

![image](https://user-images.githubusercontent.com/42154031/160629865-ccbaa445-9127-4d7f-bd1e-f40ac1bd206b.png)


```suggestion
    <GridToolbarContainer>
```

Reply:
I was for the input, I forgot to remove it